libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -800,9 +800,17 @@ bool IRGeneratorForStatements::visit(BinaryOperation const& _binOp)
 	if (TokenTraits::isCompareOp(op))
 	{
 		if (auto type = dynamic_cast<FunctionType const*>(commonType))
+			// what's this common Type class ?
+			// with this CommonType, they're not allowing ==, != operators
+			// but they're being dealt with later on
 		{
 			solAssert(op == Token::Equal || op == Token::NotEqual, "Invalid function pointer comparison!");
+			// looks like ==, != operators are not allowed on commonType function pointers
+			// will have to see if external function pointers are a type of commonType pointers only
 			solAssert(type->kind() != FunctionType::Kind::External, "External function comparison not allowed!");
+			// we want to enable comparison operators for External Functions
+			// So, this solAssert block needs to go
+			// but another question :: Why are they not allowing any ==, != comaprisons to go beyond this point
```

Comment:
Effectively, this code expects that the stack slots for the type to be just `1`. For external functions, it would be 2.

https://github.com/ethereum/solidity/blob/6849774bd758d0e984ed676eb49ed47c38aec0e7/libsolidity/ast/Types.cpp#L3116-L3121

You use the `.part(...)` to access the stack type. For an example, see

https://github.com/ethereum/solidity/blob/6849774bd758d0e984ed676eb49ed47c38aec0e7/libsolidity/codegen/ir/IRGeneratorForStatements.cpp#L1746

Two external function types are equal if both the parts / stack slots are the same.